ksql-engine/src/main/java/io/confluent/ksql/function/udf/string/SubstringKudf.java

```diff
@@ -26,13 +26,29 @@ public Object evaluate(Object... args) {
     if ((args.length < 2) || (args.length > 3)) {
       throw new KsqlFunctionException("Substring udf should have two or three input argument.");
     }
-    String string = args[0].toString();
-    long start = (Long) args[1];
-    if (args.length == 2) {
-      return string.substring((int) start);
+    
+    if(args[0]!=null && args[1]!=null) {
+    	
+    	String string = args[0].toString();
+	    int start = Integer.parseInt(args[1].toString());
+	    if (args.length == 2) {
+	    	if(start < 0 || start > string.length()) {
```

Comment:
Java's substring implementation already does this error checking for you and throws an IndexOutOfBoundsException in that case. There's no need for the ksql function to reimplement that logic. Ditto for line 42.

Reply:
it's lying around here for a while now. can somebody take care?

Reply:
@uwegeercken - I meant just catch the IndexOutOfBounds instead of reimplementing the checks.